x-pack/plugins/infra/public/containers/logs/log_analysis/api/ml_get_jobs_summary_api.ts

```diff
@@ -97,6 +97,9 @@ export const jobSummaryRT = rt.intersection([
       custom_settings: jobCustomSettingsRT,
       finished_time: rt.number,
       model_size_stats: jobModelSizeStatsRT,
+      datafeed_config: rt.partial({
+        runtime_mappings: rt.UnknownRecord,
```

Comment:
💬
I wonder if we should reassess the logs <-> ML types at some point? We had a bug somewhat recently where the ML type for errors (and the way `reasons` were sent in the payload) changed, but we didn't catch this as we replicate the types in io-ts rather than import them from ML. In that scenario things didn't break massively as it fell through to a generic error, rather than the detailed error, but it could be worse in other situations. 

We want access to the runtime encoding / decoding, but in scenarios like this we can't consume the 3rd party types without copying them, causing replication, and falling foul of things being out of sync 🤔

`rt.object` is deprecated, hence the `rt.UknownRecord`. If we think in this scenario replicating all the properties is valuable it can be changed.

Reply:
Agreed, and I think this is fine for now since we compare the runtime mappings as a whole.

Regarding duplicating the types, I wonder if there is a compromise: We could try to write type-level unit test that compare the type derived from our runtime type and the interface exported by ML. What we would gain is:

- Fail loudly on a PR that changes the ML type.
- Fail at runtime when the actual value returned by the API doesn't match the advertised type. (which has happened before)

Reply:
> We could try to write type-level unit test that compare the type derived from our runtime type and the interface exported by ML.

This sounds good, would definitely be worth a shot.